app/controllers/page_mount_controller.rb

```diff
@@ -50,13 +52,19 @@ def domains_layout
 
   private
 
-  def store_client_type_id_in_cms_session(proxy_response)
+  def store_client_type_id_in_cms_session(response, request)
     # Special Hack so CC can show exclusive products for clients
-    session[:client_type_id] = JSON.parse(proxy_response.body)['client_type_id']
-  end
-
-  def login_details_request?(request)
-    request.path == '/parties/login_header_details'
+    case request.path
+    when '/parties/login.json'
+      json = JSON.parse(response.body)
+      session[:client_id] = json['client_id']
```

Comment:
Are we using server-side session storage? Storing this information client-side could open us up to privilege escalation.